sdk/src/main/java/io/dapr/client/ObjectSerializer.java

```diff
@@ -103,13 +110,24 @@ protected ObjectSerializer() {
     }
 
     if (content.length == 0) {
-      return (T) null;
+      return null;
     }
 
     if (javaType.hasRawClass(CloudEvent.class)) {
       return (T) CloudEvent.deserialize(content);
     }
 
+    if (javaType.isTypeOrSubTypeOf(MessageLite.class)) {
```

Comment:
add a comment on why this clause is required?

is there anything that the sub type needs to implement or is it automatically implemented? for example the `parseFrom` method? Where is it coming from? Is it guaranteed that all types that satisfy this clause will have a `parseFrom` method if not we will be throwing an exception? 

Should this be documented separately?

Reply:
Let me check with another class. Not all MessageLite will have this. Since we invoke a static method, there is no interface for that.

Reply:
Keeping this class will maintain consistency with the toByteArray() from above. This is a best effort since we check for null. I fixed an issue to handle MethodNotFoundException here, so it will proceed with the usual path.